overlord/overlord.go

```diff
@@ -385,6 +409,9 @@ func (o *Overlord) Loop() {
 	if preseed {
 		o.runner.OnTaskError(preseedExitWithError)
 	}
+	if o.loopTomb == nil {
```

Comment:
can you explain the reason for these changes?

Reply:
I kept this change in a separat commit, with a description which should explain the reason:

> Do not initialize the loopTomb member until we are actually asking it to
run some routine. Otherwise the call to loopTomb.Wait() which we do when
stopping the Overlord could block forever.
> 
> This seems also to be the suggestion given in
go-tomb/tomb#21 to a similar problem.

It's a change I had to do when updating the unit tests, otherwise they were hanging forever.